packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin+AppDelegate.swift

```diff
@@ -115,4 +115,11 @@ extension AmplifyPushNotificationsPlugin {
     public func applicationDidBecomeActive(_ application: UIApplication) {
         application.registerForRemoteNotifications()
     }
+
+    public func applicationDidEnterBackground(_ application: UIApplication) {
+        // When App enters background we remove the cached launchNotification
+        // as when the App reopens after this point, there won't be a notification
+        // that launched the App.
+        launchNotification = nil
```

Comment:
Are we sure this fixes the problem? The launch notification is fetched and stored in memory by the plugin when Push configures so even if we nil it here, it will still be available on the dart side of the plugin.

Reply:
Good catch, added a Flutter API to be called from native implementation to reset the cached `launchNotification` in Dart.